app/src/main/java/com/pokerio/app/utils/GameState.kt

```diff
@@ -389,21 +389,38 @@ object GameState {
         val isThisPlayer = sha256(thisPlayer.playerID) == playerHash
 
         val player = if (isThisPlayer) thisPlayer else players.find { it.playerID == playerHash }
-        require(player != null)
+        requireNotNull(player)
 
         player.folded = true
         winningsPool += player.bet
 
         newActionCallbacks.forEach { it.value(player) }
     }
 
-    fun handleActionWon(playerHash: String) {
-        val isThisPlayer = sha256(thisPlayer.playerID) == playerHash
+    fun handleActionWon(winners: List<String>, amount: Int) {
+        winners.forEach { playerHash ->
+            val isThisPlayer = sha256(thisPlayer.playerID) == playerHash
```

Comment:
Why is this a specyfic case? We don't use anywhere else, so why won't we look just for player with expected ID?

Reply:
We do use it - remember that all we have hashes of every player apart from our own - this is how we recognize our own player (although maybe we should change it). Therefor we need this to know if we want to take our player or if we want to search for the player using the standard (non `sha256()`) search.

Reply:
Although I will put the search into a separate function, as we do it multiple times in different functions